src/services/CrawlPageProvince.ts

```diff
@@ -0,0 +1,190 @@
+
+async function crawlLinkCity() {
+  try {
+    const options = {
+      method: 'GET',
+      uri: `${URL_PROVINCES}`,
+    };
+    // winston.info(options.uri);
+    const result = await request_promise(options);
+    // winston.info('result');
+    const symbol = cheerio.load(result);
+    const dataWarehouse = [];
+    symbol(LIST_PROVINCES.DOM_LAYOUT_PROVINCES).each(function () {
+      symbol(this)
+        .find(LIST_PROVINCES.DOM_URL_PROVINCES)
+        .each(function () {
+          const dataHref = symbol(this).find('a').attr('href');
+          // winston.info(dataHref);
+          if (dataHref) {
+            dataWarehouse.push({
+              pathWareHouse: `${URL_HOME_PAGE}${dataHref}`,
+            });
+          }
+        });
+    });
+    winston.info('[crawl success data url city]');
+    return dataWarehouse;
+  } catch (err) {
+    winston.info(err);
+  }
+}
+
+async function crawlPathWareHouse() {
+  const data = [];
+  const linksCity = await crawlLinkCity();
+  // winston.info(linksCity);
+  for (const i of linksCity) {
+    try {
+      const options = {
+        method: 'GET',
+        uri: i.pathWareHouse,
+      };
+      // winston.info(options.uri);
+      const result = await request_promise(options);
+      // winston.info(result);
+      const symbol = cheerio.load(result);
+      const totalPaging = symbol(LIST_STORES.DOM_TOTAL_PAGING).text();
+      let countPaging = 1;
+      if (totalPaging === '') {
+        countPaging = 1;
+      } else {
+        // tslint:disable-next-line: radix
+        countPaging = parseInt(totalPaging);
+      }
+      winston.info(countPaging);
+
+      for (let k = 0; k < countPaging; k++) {
+        const optionsPaging = {
+          method: 'GET',
+          uri: `${options.uri}?page=${k + 1}`,
+        };
+        winston.info(optionsPaging.uri);
+        const resultPaging = await request_promise(optionsPaging);
+        const symbolPaging = cheerio.load(resultPaging);
+        symbolPaging(LIST_STORES.DOM_URL_WARE_HOUSE_).each(function () {
+          data.push({
+            url: `${URL_HOME_PAGE}${symbolPaging(this).find('a').attr('href')}`,
+            status: 0,
+          });
+        });
+        // winston.info('dataPage');
+        fs.writeFile('pathDetailsWareHouse.json', JSON.stringify(data), (err) => {
+          if (err) throw err;
+          winston.info('done !');
+        });
+      }
+    } catch (error) {
+      winston.info(error);
+    }
+  }
+  winston.info('[crawl success data url ware house]');
+  return data;
+}
+
+async function detailPageWarehouse() {
+  try {
+    // Check if the file exists or not
+    const path = 'pathDetailsWareHouse.json';
+    if (!fs.existsSync(path)) {
+      // Create file;
+      fs.writeFile('pathDetailsWareHouse.json', '', (err) => {
+        if (err) throw err;
+      });
+    }
+
+    // Read file json
+    let dataPathWareHouse = await fsPromises.readFile('pathDetailsWareHouse.json', 'utf8');
```

Comment:
đã có cái biến https://github.com/d10cn2btt/typescript-nodejs/pull/6/files#diff-8b4205d96645d6ac46e07d0f460aadcb4d3d88dd44db5d504a0358e10463fde4R96
rồi thì sao k dùng ?

Reply:
em chưa hiểu ý anh ạ, đây là em tạo ra 1 file pathDetailsWareHouse.json để tránh bị bị lỗi khi đọc file 
    // Read file json
    let dataPathWareHouse = await fsPromises.readFile('pathDetailsWareHouse.json', 'utf8');
 em khởi tạo file đó bên trong không có dữ liệu

Reply:
```suggestion
      fs.writeFile(filePathDetailsWareHouse, '', (err) => {
        if (err) throw err;
      });
    }

    // Read file json
    let dataPathWareHouse = await fsPromises.readFile(filePathDetailsWareHouse, 'utf8');
```

Thế này thì sao em ?

Reply:
em đã đặt tên biến rồi ạ.
const fileNameUrlPageProvinces = 'urlPageProvinces.json';
const fileNameUrlDetailsWareHouse = 'urlDetailsWarehouse.json';
const fileNameOutput = 'output.json';

Reply:
.